core/src/com/cloud/info/ConsoleProxyInfo.java

```diff
@@ -55,6 +57,9 @@ public ConsoleProxyInfo(boolean sslEnabled, String proxyIpAddress, int port, int
                 proxyImageUrl += ":" + this.proxyUrlPort;
         } else {
             proxyAddress = proxyIpAddress;
+            if (!Strings.isNullOrEmpty(consoleProxyUrlDomain)) {
```

Comment:
you can use directly `org.apache.commons.lang3.StringUtils.isNotBlank(CharSequence)`, then you do not need the `!`. It also catches "blanks" values, but I am not sure if they are possible here.

Reply:
@rafaelweingartner I think there is a divide in preference to use commons.lang or guava, we already have a lot of code that uses guava and things like `Strings.isNullOrEmpty`. I prefer using the guava utility/methods, also StringUtils.isNotBlank internally would do more or less the same kind of checks. I'm okay to change, but then we need a java style/doc on which methods to prefer in which case we can deprecate the use of one style over the other.

Reply:
No problem, but jut to be clear, that method (`Strings.isNullOrEmpty`) does not address the blank case.

Reply:
@rafaelweingartner I'm not sure but last time I checked, the `isNullOrEmpty` did check for emptiness (blank) of string (https://google.github.io/guava/releases/snapshot-jre/api/docs/com/google/common/base/Strings.html#isNullOrEmpty-java.lang.String-)

Reply:
Well the java dos they just mention empty, and not blank. I checked their code, and they do not check for blank.

Test code:
```
    public static void main(String[] args) {
        System.out.println("Using Guava Strings suite");
        System.out.println("Result for empty: " + Strings.isNullOrEmpty(""));
        System.out.println("Result for null " + Strings.isNullOrEmpty(null));
        System.out.println("Result for blank: " + Strings.isNullOrEmpty("       "));
        System.out.println();
        System.out.println("Using Apache commns lang");
        System.out.println("Result for empty: " + StringUtils.isBlank(""));
        System.out.println("Result for null " + StringUtils.isBlank(null));
        System.out.println("Result for blank: " + StringUtils.isBlank("       "));
    }
```

Result:
```
Using Guava Strings suite
Result for empty: true
Result for null true
Result for blank: false

Using Apache commns lang
Result for empty: true
Result for null true
Result for blank: true
```

Reply:
Thanks for testing and sharing @rafaelweingartner used `StringUtils` now!